sync/class.jetpack-sync-module-posts.php

```diff
@@ -318,98 +287,111 @@ function filter_post_content_and_add_links( $post_object ) {
 	}
 
 	public function save_published( $new_status, $old_status, $post ) {
-		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
-			$this->just_published[ $post->ID ] = true;
+		if ( ! isset( $this->sync_items[ $post->ID ] ) ) {
+			$this->sync_items[ $post->ID ] = new Jetpack_Sync_Item( 'save_post' );
 		}
-
-		$this->previous_status[ $post->ID ] = $old_status;
+		$sync_item = $this->sync_items[ $post->ID ];
+		$is_just_published = 'publish' === $new_status && 'publish' !== $old_status;
+		$sync_item->set_state_value( 'is_just_published', $is_just_published );
+		$sync_item->set_state_value( 'previous_status', $old_status );
```

Comment:
Should we consider having explicit get_state(), set_state(), get_previous_state(), and set_previous_state() methods on the sync object class?